sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -257,16 +261,33 @@ class FileStreamSource(
    * equal to `end` and will only request offsets greater than `end` in the future.
    */
   override def commit(end: Offset): Unit = {
-    // No-op for now; FileStreamSource currently garbage-collects files based on timestamp
-    // and the value of the maxFileAge parameter.
+    val logOffset = FileStreamSourceOffset(end).logOffset
+
+    if (sourceOptions.cleanSource != CleanSourceMode.NO_OP) {
+      val files = metadataLog.get(Some(logOffset), Some(logOffset)).flatMap(_._2)
+      val validFileEntities = files.filter(_.batchId == logOffset)
+      logDebug(s"completed file entries: ${validFileEntities.mkString(",")}")
+      sourceOptions.cleanSource match {
+        case CleanSourceMode.ARCHIVE =>
+          validFileEntities.foreach(sourceCleaner.archive)
+
+        case CleanSourceMode.DELETE =>
```

Comment:
Should the delete operation update the commit to remove the file as well if possible? Asking because if a file gets deleted and then another file with the same name was uploaded, it wont get processed.

Reply:
Hmm... not sure how it works with current file stream source. I'd rather keep compatibility with current behavior so I need to check it first.

Reply:
Without this change the file which uploaded with the same name not get processed again. It may cause issues with exactly once...

Reply:
I guess I don't touch the mechanism how file source deals with files, which means it doesn't change the behavior. (Please correct me if I'm missing here.) That's why I need to check how current file stream source deals with new file with same name. If the behavior of current master and this patch are same and it doesn't work as we expect/want, addressing it is beyond this PR and better to be filed and addressed separately.

Another thing to note is, we need to also consider that some options like `fileNameOnly` which make things more complicated.

Reply:
Now I'm seeing #23782 to address this: if #23782 doesn't need to remove the completed files from commit, logically it should be same here. So I'd rather rely on #23782 and have a discussion there.

Reply:
@HeartSaVioR reading back my comment from yesterday is a bit... well not really clean :)

> It may cause issues with exactly once...

references the same thing what you've mentioned on #23782 and agree with you.

Reprocessing files with the same name may cause several issues. What if file being archived and some producer is trying to append it, etc...
I've left a more detailed comment on the mentioned PR. There I've added how streaming dealing with incoming files.